### PR TITLE
Diescalo/android split

### PR DESCRIPTION
**Pull request by diescalo**

### CODE REVIEW

File: library/src/main/java/com/digi/xbee/api/AbstractXBeeDevice.java [hunk 1]
```diff
@@ -2738,7 +2701,23 @@ public void executeParameter(String parameter) throws TimeoutException, XBeeExce
 	 */
 	@Override
 	public String toString() {
-		return connectionInterface.toString();
+		String id = getNodeID() == null ? "" : getNodeID();
+		String addr64 = get64BitAddress() == null || get64BitAddress().equals(XBee64BitAddress.UNKNOWN_ADDRESS) ? 
+				"" : get64BitAddress().toString();
+		
+		if (id.length() == 0 && addr64.length() == 0)
+			return super.toString();
```
Comment (rubenmoral) [inline library/src/main/java/com/digi/xbee/api/AbstractXBeeDevice.java]: This should be `return connectionInterface.toString();` (and the same below).

File: library/src/main/java/com/digi/xbee/api/AbstractXBeeDevice.java [hunk 2]
```diff
@@ -3091,4 +3070,1839 @@ public void updateBluetoothPassword(String newPassword) throws TimeoutException,
...
+	 *                       connection.
+	 * 
+	 * @see #close()
+	 * @see #isOpen()
+	 */
+	protected void open() throws XBeeException {
+		logger.info(toString() + "Opening the connection interface...");
+		
+		if (isRemote())
+			throw new RuntimeException("Remote devices cannot be open.");
```
Comment (rubenmoral) [inline library/src/main/java/com/digi/xbee/api/AbstractXBeeDevice.java]: I would throw here an `OperationNotSupportedException`.

File: library/src/main/java/com/digi/xbee/api/connection/ConnectionType.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+ * Enumerates the different connection types.
+ * 
+ * @since 1.3.0
+ */
+public enum ConnectionType {
+	
+	// Enumeration types.
+	SERIAL(0, "Serial connection"),
+	BLUETOOTH(1, "Bluetooth connection"),
+	UNKNOWN(3, "CoAP");
```
Comment (rubenmoral) [inline library/src/main/java/com/digi/xbee/api/connection/ConnectionType.java]: Wrong description, it should be `"Unknown"`.

Comment (rubenmoral) [reply library/src/main/java/com/digi/xbee/api/connection/ConnectionType.java]: And why 3 as ID? Better 0xFF?

File: library/src/main/java/com/digi/xbee/api/connection/ConnectionType.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	static {
+		for (ConnectionType connectionType:values())
+			lookupTable.put(connectionType.getID(), connectionType);
+	}
+
+	/**
+	 * Class constructor. Instantiates a new {@code ConnectionType} enumeration
+	 * entry with the given parameters.
+	 *
+	 * @param id IP connection type ID.
```
Comment (rubenmoral) [inline library/src/main/java/com/digi/xbee/api/connection/ConnectionType.java]: Remove IP here and below.

File: library/src/main/java/com/digi/xbee/api/connection/ConnectionType.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+	/**
+	 * Retrieves the connection type for the given ID.
+	 *
+	 * @param id ID to retrieve the connection type.
+	 *
+	 * @return The connection type associated with the given ID.
+	 */
+	public static ConnectionType get(int id) {
+		return lookupTable.get(id);
```
Comment (rubenmoral) [inline library/src/main/java/com/digi/xbee/api/connection/ConnectionType.java]: If we are not returning `UNKNOWN` if the given ID is not in the table, should we keep that entry?

Comment (diescalo) [reply library/src/main/java/com/digi/xbee/api/connection/ConnectionType.java]: I'll change it to return `UNKNOWN` if the ID is not in the list.

**Pull request merged**